target/linux/ath79/mikrotik/base-files/etc/board.d/02_network

```diff
@@ -21,6 +21,10 @@ ath79_setup_interfaces()
 	mikrotik,routerboard-wapr-2nd)
 		ucidef_set_interface_lan "eth0"
 		;;
+	mikrotik,routerboard-962uigs-5hact2hnt)
+		ucidef_add_switch "switch0" \
+			"0@eth0" "2:lan" "3:lan" "4:lan" "5:lan" "1:wan"
+		;;
```

Comment:
This does not define lan/wan: are the MAC addresses correct once booted?
How about port numbering  in Luci: is it correct as well?
Also what is eth1 used for since it does not appear here?

Reply:
Mac address is ok for LAN port and WAN (eth0.2). Is it SFP port eth0.1 or eth1?
```
  1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
      link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
      inet 127.0.0.1/8 scope host lo
         valid_lft forever preferred_lft forever
      inet6 ::1/128 scope host
         valid_lft forever preferred_lft forever
  2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP qlen 1000
      link/ether aa:bd:2b:e9:01:25 brd ff:ff:ff:ff:ff:ff
      inet6 fe80::a8bd:2bff:fee9:125/64 scope link
         valid_lft forever preferred_lft forever
  3: eth1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
      link/ether 42:ee:c6:0b:3a:28 brd ff:ff:ff:ff:ff:ff
  6: wlan1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
      link/ether some:mac:address:b2 brd ff:ff:ff:ff:ff:ff
  7: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
      link/ether some:mac:address:ac brd ff:ff:ff:ff:ff:ff
      inet 192.168.88.1/24 brd 192.168.88.255 scope global br-lan
         valid_lft forever preferred_lft forever
      inet6 fded:7949:5479::1/60 scope global noprefixroute
         valid_lft forever preferred_lft forever
      inet6 fe80::764d:28ff:fecf:8cac/64 scope link
         valid_lft forever preferred_lft forever
  8: eth0.1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP qlen 1000
      link/ether some:mac:address:ac brd ff:ff:ff:ff:ff:ff
  9: eth0.2@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
      link/ether some:mac:address:ab brd ff:ff:ff:ff:ff:ff
      inet SOME IP ADDRESS.202/26 brd SOME IP ADDRESS.255 scope global eth0.2
         valid_lft forever preferred_lft forever
      inet6 fe80::764d:28ff:fecf:8cab/64 scope link
         valid_lft forever preferred_lft forever
  10: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP qlen 1000
      link/ether some:mac:address:b1 brd ff:ff:ff:ff:ff:ff
      inet6 fe80::764d:28ff:fecf:8cb1/64 scope link
         valid_lft forever preferred_lft forever
```

Reply:
Hi Daniel. Following the output where you can see that both eth0 and eth1 have MAC different from the "real one" (the one written on the sticker, that starts with b8:69:f4:35:d6:74 and ends with b8:f4:35:d6:7b for WAN1).
I don't know exactly what this means and, also, what I have to do with the four lines of code you've written for "target/linux/ath79/mikrotik/base-files/etc/board.d/02_network"
If you think it's useful, I could reload the RouterOS on the hAP ac to write down the exact mac for the SFP module, even if I suppose it's b8:69:f4:35:d6:79.
Let me know if I could be useful for some testing. Thanks

1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP qlen 1000
    link/ether 02:30:fc:e2:ea:26 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::30:fcff:fee2:ea26/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether f6:dd:03:43:fe:69 brd ff:ff:ff:ff:ff:ff
5: wlan0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether b8:69:f4:35:d6:7a brd ff:ff:ff:ff:ff:ff
6: wlan1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether b8:69:f4:35:d6:7b brd ff:ff:ff:ff:ff:ff
7: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether b8:69:f4:35:d6:75 brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.1/24 brd 192.168.1.255 scope global br-lan
       valid_lft forever preferred_lft forever
    inet6 fd45:93f:f7fb::1/60 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fe80::ba69:f4ff:fe35:d675/64 scope link 
       valid_lft forever preferred_lft forever
8: eth0.1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP qlen 1000
    link/ether b8:69:f4:35:d6:75 brd ff:ff:ff:ff:ff:ff
9: eth0.2@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether b8:69:f4:35:d6:74 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::ba69:f4ff:fe35:d674/64 scope link 
       valid_lft forever preferred_lft forever

Reply:
On my hap AC, on sticker `E01` is same as on port `eth0.2@eth0` and what is on `W01` is what is set in `wlan1`.

Reply:
OK so the VLANs have the correct MAC addresses but the root devices (eth0/eth1) do not. I don't know if that's ok. @adschm, what do you think?

Reply:
`eth1` is SFP, this worked with 2 different GLC-T modules (which didn't work with RouterOS). I don't have any fibre transceivers to test with. I'll get it out this weekend & check the MAC address situation incl. SFP.

I used other devices (mpc85xx WDR4900) that have correct MAC address on VLAN ifaces but not on root dev so I think there is precedent there.

Reply:
Hi all. I've made a recap, hoping it could be useful:
My hAP ac with RouterOS immediatly after installation:

/interface ethernet> print
Flags: X - disabled, R - running, S - slave 
 #    NAME              MTU MAC-ADDRESS       ARP             SWITCH           
 0    ether1           1500 B8:69:F4:35:D6:74 enabled         switch1          
 1 RS ether2           1500 B8:69:F4:35:D6:75 enabled         switch1          
 2  S ether3           1500 B8:69:F4:35:D6:76 enabled         switch1          
 3  S ether4           1500 B8:69:F4:35:D6:77 enabled         switch1          
 4  S ether5           1500 B8:69:F4:35:D6:78 enabled         switch1          
 5 RS sfp1             1500 B8:69:F4:35:D6:79 enabled        

The sfp1 is equipped with a original S-31DLC20D (Single Mode, 20Km) and is connected to an identical module on another Mikrotik:
/interface ethernet> monitor 5
                      name: sfp1
                    status: link-ok
          auto-negotiation: done
                      rate: 1Gbps
               full-duplex: yes
           tx-flow-control: no
           rx-flow-control: no
               advertising: 
  link-partner-advertising: 
        sfp-module-present: yes
               sfp-rx-loss: no
                  sfp-type: SFP-or-SFP+
        sfp-connector-type: LC
       sfp-link-length-9um: 20000m
           sfp-vendor-name: MikroTik
    sfp-vendor-part-number: S-31DLC20D
       sfp-vendor-revision: 2.0
         sfp-vendor-serial: OL2110251169
    sfp-manufacturing-date: 21-10-25
            sfp-wavelength: 1310nm
           sfp-temperature: 40C
        sfp-supply-voltage: 3.197V
       sfp-tx-bias-current: 19mA
              sfp-tx-power: -6.278dBm
              sfp-rx-power: -7.649dBm
           eeprom-checksum: good
                    eeprom: 0000: 03 04 07 00 00 00 02 12  10 01 00 03 0d 00 14 c8  ........ ........
                            0010: 00 00 00 00 4d 69 6b 72  6f 54 69 6b 20 20 20 20  ....Mikr oTik    
                            0020: 20 20 20 20 00 00 00 00  53 2d 33 31 44 4c 43 32      .... S-31DLC2
                            0030: 30 44 20 20 20 20 20 20  32 2e 30 20 05 1e 00 39  0D       2.0 ...9
                            0040: 00 1a 00 00 4f 4c 32 31  31 30 32 35 31 31 36 39  ....OL21 10251169
                            0050: 20 20 20 20 32 31 31 30  32 35 20 20 68 b0 01 b5      2110 25  h...
                            0060: 00 00 08 0d 45 0d cc 22  34 fd 12 e8 02 fa 8c 6a  ....E.." 4......j
                            0070: 20 da 2c 00 00 00 00 00  00 00 00 00 97 cc 93 11   .,..... ........
                            0080: 64 00 d8 00 5f 00 dd 00  8c a0 6d 60 88 b8 71 48  d..._... ..m`..qH
                            0090: 9c 40 01 f4 88 b8 03 e8  18 a6 03 e8 13 94 04 eb  .@...... ........
                            00a0: 4d f1 00 14 27 10 00 28  00 00 00 00 00 00 00 00  M...'..( ........
                            00b0: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ........ ........
                            00c0: 00 00 00 00 3f 80 00 00  00 00 00 00 01 00 00 00  ....?... ........
                            00d0: 01 00 00 00 01 00 00 00  01 00 00 00 00 00 00 19  ........ ........
                            00e0: 29 16 7c 68 26 d6 09 10  08 fc ff ff ff ff 02 ff  ).|h&... ........
                            00f0: 00 40 ff ff 00 40 ff ff  ff ff ff ff ff ff ff 00  .@...@.. ........

/interface wireless> print      
Flags: X - disabled, R - running 
 0    name="wlan1" mtu=1500 l2mtu=1600 mac-address=B8:69:F4:35:D6:7B 
      arp=enabled interface-type=Atheros AR9300 mode=ap-bridge 
      ssid="MikroTik-35D67B" frequency=auto band=2ghz-b/g/n 
      channel-width=20/40mhz-Ce secondary-channel="" scan-list=default 
      wireless-protocol=802.11 vlan-mode=no-tag vlan-id=1 wds-mode=disabled 
      wds-default-bridge=none wds-ignore-ssid=no bridge-mode=enabled 
      default-authentication=yes default-forwarding=yes default-ap-tx-limit=0 
      default-client-tx-limit=0 hide-ssid=no security-profile=default 
      compression=no 

 1    name="wlan2" mtu=1500 l2mtu=1600 mac-address=B8:69:F4:35:D6:7A 
      arp=enabled interface-type=Atheros AR9888 mode=ap-bridge 
      ssid="MikroTik-35D67A" frequency=auto band=5ghz-a/n/ac 
      channel-width=20/40/80mhz-Ceee secondary-channel="" scan-list=default 
      wireless-protocol=802.11 vlan-mode=no-tag vlan-id=1 wds-mode=disabled 
      wds-default-bridge=none wds-ignore-ssid=no bridge-mode=enabled 
      default-authentication=yes default-forwarding=yes default-ap-tx-limit=0 
      default-client-tx-limit=0 hide-ssid=no security-profile=default 
      compression=no 

The two MAC Addresses reported on the sticker at the bottom of may hAP ac are the one that corresponds to interface "ether1" and interface "wlan1"
After taking these printouts I reinstalled the Daniel build and the output is exactly as before:

# ip addr show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP qlen 1000
    link/ether a2:01:f5:27:16:7c brd ff:ff:ff:ff:ff:ff
    inet6 fe80::a001:f5ff:fe27:167c/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 52:c3:e4:c1:9a:9e brd ff:ff:ff:ff:ff:ff
5: wlan0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether b8:69:f4:35:d6:7a brd ff:ff:ff:ff:ff:ff
6: wlan1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether b8:69:f4:35:d6:7b brd ff:ff:ff:ff:ff:ff
7: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether b8:69:f4:35:d6:75 brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.1/24 brd 192.168.1.255 scope global br-lan
       valid_lft forever preferred_lft forever
    inet6 fd02:f0d3:267b::1/60 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fe80::ba69:f4ff:fe35:d675/64 scope link 
       valid_lft forever preferred_lft forever
8: eth0.1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP qlen 1000
    link/ether b8:69:f4:35:d6:75 brd ff:ff:ff:ff:ff:ff
9: eth0.2@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether b8:69:f4:35:d6:74 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::ba69:f4ff:fe35:d674/64 scope link 
       valid_lft forever preferred_lft forever

The MAC Addresses of eth0 and eth1 are unknown (to me) while eth0.1, eth0.2 and wlan0 correspond to the ones of RouterOS.
In Luci->Network->Switch only the 5 ethernet ports are visibile.

Looking around, I've discovered a similar issue in the official release of 21.02 for Mikrotik hEX S (device page: https://openwrt.org/toh/mikrotik/rb760igs):
"Fiber SFP is broken in 21.02. This patch in master fixes it, but it is not backported to 21.02, so you have to use snapshot builds if you use fiber SFP modules. Copper modules should work in 21.02."
I don't know if this is applicable also to hAP ac, but I would like to use the sfp port with fiber module, if possibile (the link is: https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=54ed38d5ea)